client/src/components/MultiJurisdictionAudit/AASetup/Contests/index.test.tsx

```diff
@@ -258,6 +258,66 @@ describe('Audit Setup > Contests', () => {
     })
   })
 
+  it('it should skip to next stage when opportunistic contest form is clean and not touched', async () => {
+    const { findByText, getByText } = render(
+      <Contests
+        auditType="BALLOT_POLLING"
+        locked={false}
+        isTargeted={false}
+        nextStage={nextStage}
+        prevStage={prevStage}
+      />
+    )
+
+    await findByText('Opportunistic Contests')
+    fireEvent.click(getByText('Save & Next'), { bubbles: true })
+    await waitFor(() => {
+      expect(nextStage.activate).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  it('it should not skip to next stage when targeted contest form is clean and not touched', async () => {
+    const { findByText, getByText } = render(
+      <Contests
+        auditType="BALLOT_POLLING"
+        locked={false}
+        isTargeted
+        nextStage={nextStage}
+        prevStage={prevStage}
+      />
+    )
+
+    await findByText('Target Contests')
+    fireEvent.click(getByText('Save & Next'), { bubbles: true })
+    await waitFor(() => {
+      expect(nextStage.activate).toHaveBeenCalledTimes(0)
+    })
```

Comment:
will this check work? since you're checking to see if `nextStage.activate` has been called 0 times, is it possible that this check will pass before the button handler has time to get called? if so, then you might want to check something else - e.g. that the button is disabled or that an error message pops up. if you think this check will work, then i would remove the `await waitFor`, since that implies that there's some asynchronous action being waited for

Reply:
@jonahkagan Oh interesting, good catch. :thinking: It should bring up the error messages from the fields and I can use that.